runtime/src/chpl-topo.c

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright 2017-2017 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ * 
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * 
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+// Compute node topology support
+//
+#include "chplrt.h"
+
+#include "chpl-align.h"
+#include "chplsys.h"
+#include "chpl-topo.h"
+#include "chpltypes.h"
+#include "error.h"
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+
+#ifdef CHPL_HAS_HWLOC
+//
+// We have hwloc, so provide a real implementation of the topology
+// interface.
+//
+#include "hwloc.h"
+
+#ifdef DEBUG
+// note: format arg 'f' must be a string constant
+#define _DBG_P(f, ...)                                                  \
+        do {                                                            \
+          printf("%s:%d" f "\n", __FILE__, __LINE__, ## __VA_ARGS__);   \
+        } while (0)
+#else
+#define _DBG_P(f, ...)
+#endif
+
+static hwloc_topology_t topology;
+
+static const struct hwloc_topology_support* topoSupport;
+
+static int topoDepth;
+
+static int numaLevel;
+static int numNumaDomains;
```

Comment:
On second thought, I should probably stick with hbw_available() which I've been using, as it's a better indicator for my particular purpose.  No need to have a getter for numNumaDomains unless there is another use for it.